src/shillelagh/lib.py

```diff
@@ -567,7 +567,12 @@ def find_adapter(
     for adapter in adapters:
         key = adapter.__name__.lower()
         kwargs = adapter_kwargs.get(key, {})
-        supported: Optional[bool] = adapter.supports(uri, fast=True, **kwargs)
+
+        try:
+            supported: Optional[bool] = adapter.supports(uri, fast=True, **kwargs)
+        except Exception:  # pylint: disable=broad-except
+            supported = False
```

Comment:
Unrelated, but this was always a nuisance when running tests, so I decided to go ahead and fix this.